zetaclient/chains/evm/observer/inbound.go

```diff
@@ -672,11 +743,15 @@ func (ob *Observer) BuildInboundVoteMsgForDepositedEvent(
 			Msgf("thank you rich folk for your donation! tx %s chain %d", event.Raw.TxHash.Hex(), ob.Chain().ChainId)
 		return nil
 	}
+
+	// convert erc20Address to asset in foreign coin store to avoid checksum mismatch
```

Comment:
```suggestion
	// convert erc20Address to asset in foreign coin store to avoid checksum mismatch
	// TODO: remove once the checksum conversion is fixed in the protocol
	// https://github.com/zeta-chain/node/issues/3274
```